DecipherPD LUHMES EPIC Array Runs - Methylation and Expression Network Analysis
================================

##### Analyst: Samantha Schaffner
##### Date: Sept 30, 2020

This project investigates the impact of alpha-synuclein genotype on DNA methylation and hydroxymethylation in human dopaminergic neurons, as part of the DecipherPD grant. The model used was Lund human mesencephalic (LUHMES) cells, a fetal midbrain cell line, that were differentiated into dopaminergic neurons.

Tiago Outerio's group (Gottingen, Germany) created LUHMES overexpressing wild type or WT mutant human alpha-synuclein (SNCA) as well as GFP. DNA was extracted from 7 replicates of control cells, 8 replicates of SNCA-overexpressing (SNCA-OE) cells, and 8 replicates of WT cells. Each sample was split into two for bisulfite and oxidative bisulfite conversion, and these were run on the EPIC array. DNA methylation was caluclated using the oxidative bisulfite-converted samples, while DNA hydroxymethylation was calculated by subtracting the oxidative bisulfite-converted beta values (mC) from the bisulfite-converted beta values (hmC + mC).

This script uses differential methylation and differential expression data comparing SNCA-OE and control cells. Using the Significance-based Modules Integrating the Transcriptome and Epigenome (SMITE) package (Wijetunga et al, 2017), I will create an object to store the methylation and expression data.

This object will be analyzed using the SMITE workflow and annotating gene scores to a protein-protein network.

#Libraries and functions
```{r libraries, eval=F}
setwd("~/")
options(stringsAsFactors = FALSE)
library(SMITE)
library(dplyr)
library(hexbin)
library(biomaRt)
library(igraph)
library(Rsamtools)
source("~/KoborLab/kobor_space/sschaffner/LUHMES_201806/clean_scripts/plotModule2.R") #I altered the plotModule function included in the SMITE package ("plotModule2.R") after I noticed the original function was a bit misleading. It was set to plot any DNAm value of 1 as "high", anything significant other than 1 as "low", and anything non-significant as "medium." So I tweaked some parameters to plot any delta beta < 0 as "Hypo" for hypomethylated, and any delta beta > 0 as "Hyper" for hypermethylated. I also renamed the "medium" to "NS" for non-significant in both the DNAm and expression categories.
source("~/KoborLab/kobor_space/sschaffner/DecipherPD_RRBS/SNCA_OE/Hippocampus/12m_EE/multi_omic/annotateModification2.R") #This fixes a bracket typo in a conditional loop (line 106) where if there is only one CpG sites per methylation feature, the effect for that feature should be returned (instead of attempting to weight the effect, which returns NA). Also added a condition where if all the p-values for a feature are 1, they will be replaced with 0.9999999999. Otherwise if all p-values are 1, all weights will be 0, resulting in NA for the weighted effect.
```

#Load and curate DNA methylation data
The output dataframe should have each row representing one CpG site with columns containing chromosome, start, stop, effect size (delta beta), and adjusted p-value (FDR) in that order.

*For site-specific data, start/stop are the same (CpG position).
```{r DNAm data, eval=F}
#Load DNA methylation data - data frame containing chr, start, stop, effect size (delta beta), and adjusted p-value (FDR)
WT_DMPs <- read.csv("~/KoborLab/kobor_space/sschaffner/LUHMES_201806/clean_scripts/DMPs_WT_06182020.csv")

methylation <- WT_DMPs[,c("chr", "pos", "pos", "DB", "adj.P.Val")]
colnames(methylation)[2:3] <- c("start", "stop")

#Remove NAs and pvals == 0 (replace with minimum pval)
#Replace p-values
zeros <- methylation %>% filter(adj.P.Val == 0)
nrow(zeros) #0 - no zero pvals
#Remove NAs
methylation <- methylation[complete.cases(methylation$adj.P.Val),]

#Save
save(methylation, file="~/KoborLab/kobor_space/sschaffner/LUHMES_201806/clean_scripts/methylation_WT_06232020.RData")
```

#Load and curate RNA-seq data
The output dataframe should have gene names as rownames, effect size (log2FC) in the first column, and p-value in the second column.
```{r RNAseq, eval=F}
expression <- read.csv("~/KoborLab/kobor_space/sschaffner/LUHMES_201806/clean_scripts/Paiva_RNAseq_WT.csv", sep=",", skip=1)
#Remove values without a gene symbol
expression <- expression[complete.cases(expression$gene_name),]
#Replace zero p-values
min(expression$padj) # 0
min_padj <- min(expression[expression$padj>0,]$padj)
expression[expression$padj==0,]$padj <- min_padj
summary(expression$padj) #all pvals are >0, maximum 1

#Subset to relevant columns
expression <- expression[-which(duplicated(expression$gene_name)), ]
rownames(expression) <- expression$gene_name
expression <- expression[,c("log2FoldChange", "padj")]
save(expression, file="~/KoborLab/kobor_space/sschaffner/LUHMES_201806/clean_scripts/expression_WT.RData")
```

#Load bed file - expression
Since the RNA-seq is annotated with Ensembl, we will use Ensembl gene annotations from hg19. The BED file can be exported from the UCSC genome table browser (http://genome.ucsc.edu/cgi-bin/hgTables); here it was saved as "hg19_genes" before opening in R.
```{r bed file, eval=F}
#hg19_bed <- read.table("hg19_genes",header = FALSE, sep="\t",stringsAsFactors=FALSE, quote="")

#Need to convert Ensemble transcripts to genes
#Use the biomaRt package
#my_mart = useMart("ENSEMBL_MART_ENSEMBL", dataset="hsapiens_gene_ensembl", host="www.ensembl.org")
#hg19_genes <- getBM(c("ensembl_gene_id", "ensembl_transcript_id", "hgnc_symbol"), filters="ensembl_transcript_id", values=hg19_bed[,4], mart= my_mart)
#hg19_genes <- hg19_genes[-which(duplicated(hg19_genes$ensembl_transcript_id)), ]

#nrow(hg19_genes) #189723
#nrow(hg19_bed) #204940
#hg19_bed <- hg19_bed[(hg19_bed[,4] %in% hg19_genes$ensembl_transcript_id),] #subset to genes/transcripts with available annotations
#hg19_bed <- hg19_bed %>% arrange(V4)
#hg19_genes <- hg19_genes %>% arrange(ensembl_transcript_id)
#identical(hg19_genes$ensembl_transcript_id, hg19_bed$V4) #TRUE
#hg19_bed[,4] <- hg19_genes$hgnc_symbol #Replace transcript ID with gene ID
#save(hg19_bed, file="hg19_bed.RData")

load("~/KoborLab/kobor_space/sschaffner/LUHMES_201806/clean_scripts/hg19_bed.RData")
```

#Load bed file - H3K4me1 (enhancer track)
Additonal parameters other than DNA methylation and expression are optional inputs to SMITE. Here, I used H3K4me1 ChIP-seq data to represent enhancers.

LUHMES ChIP-seq was performed by Zinah Wassouf (Outeiro lab, University of Goettingen/DZNE, Goettigen, Germany).

Ensure the output dataframe contains chromosome, start, and stop in columns 1-3, respectively.
```{r enhancer, eval=F}
enhancer <- read.csv("~/KoborLab/kobor_space/sschaffner/LUHMES_201806/ChIPseq/H3K4me1/Peaks_H3K4me1_Consensus.csv")

#use enhancers from GFP group
#0 indicates no evidence for enhancers; remove rows with a scores of 0 in GFP
enhancer <- enhancer[enhancer$GFP>0,]
```

#Integrate data
Note: check the help functions for annotations to ensure input data frames are structured correctly (the vignette does not make this clear).
```{r integrate data, eval=F}
#Load in hydroxymethylation data
WT_DHMPs <- read.csv("~/KoborLab/kobor_space/sschaffner/LUHMES_201806/clean_scripts/DHMPs_WT_06182020.csv")

hydroxymethylation <- WT_DHMPs[,c("chr", "pos", "pos", "DB", "adj.P.Val")]
colnames(hydroxymethylation)[2:3] <- c("start", "stop")
summary(hydroxymethylation$adj.P.Val)

#Create PvalueAnnotation
#"other_tss_distance" argument can take unique values for each type of "other_data"; left to default 10,000 bp due to heterogeneity
#This considers promoters +/- 1000 bp from TSS, genes bodies +1000bp from TSS - TES, and enhancers +/- 5000bo from H3K4me1 peaks
LUHMES_annotation <- makePvalueAnnotation(data = hg19_bed, other_data=list(enhancer=enhancer), gene_name_col = 4, promoter_upstream_distance = 1000, promoter_downstream_distance = 1000, other_tss_distance = 5000)

#Load in expression data
LUHMES_annotation <- annotateExpression(pvalue_annotation = LUHMES_annotation, expr_data = expression, effect_col = 1, pval_col = 2)

#Load in methylation data
#Weighted by pvalue
LUHMES_annotation <- annotateModification2(LUHMES_annotation, methylation, weight_by_method = "Stouffer", weight_by = c(promoter = "pval", body = "pval", enhancer="pval"), verbose = TRUE, mod_corr = FALSE, mod_type = "methylation")

#Load in hydroxy data
LUHMES_annotation <- annotateModification2(LUHMES_annotation, hydroxymethylation, weight_by_method = "Stouffer", weight_by = c(promoter = "pval", body = "pval", enhancer="pval"), verbose = TRUE, mod_corr = FALSE, mod_type = "hydroxy")

#View loaded data
head(extractExpression(LUHMES_annotation))
extractModification(LUHMES_annotation, mod_type = "methylation")
head(extractModSummary(LUHMES_annotation))
```

#Adjusting values and scoring
```{r adjust and score, eval=F}
#Create a PvalueObject (combines all data)
#Specify effect directions for each modification-context pairing; opposite directions will be penalized
LUHMES_annotation <- makePvalueObject(pvalue_annotation= LUHMES_annotation, effect_directions = c(methylation_promoter = "bidirectional", methylation_body = "bidirectional", methylation_enhancer="bidirectional", hydroxy_promoter="bidirectional", hydroxy_body="bidirectional", hydroxy_enhancer="bidirectional"))

#Plot density of p-values
png("~/KoborLab/kobor_space/sschaffner/LUHMES_201806/clean_scripts/plotDensityPval_WT_092020.png")
plotDensityPval(pvalue_annotation = LUHMES_annotation, ref = "expression_pvalue")
dev.off()
```
![WT raw p-value densities](/home/BCRICWH.LAN/sschaffner/KoborLab/kobor_space/sschaffner/LUHMES_201806/clean_scripts/plotDensityPval_WT_092020.png)

#Normalization
Creating the PvalueAnnotation object altered the p-value distibutions for methylation and hydroxymethylation input data as p-values were combined across genomic intervals. Applying a normalization procedure ensures that p-values for each modification have a similar range to each other, so as not to bias which types of modification-context pairings emerge in the modules later on. The "rescale" method first applies a logit transformation, then shifts the p-values within an approximately normal distribution, without affected the relative order of p-values within each modification-context pairing.
```{r normalize, eval=FALSE}
LUHMES_annotation_rescaled <- normalizePval(LUHMES_annotation, ref="expression", method="rescale")
```
![WT rescaled normalized p-value densities](/home/BCRICWH.LAN/sschaffner/KoborLab/kobor_space/sschaffner/LUHMES_201806/clean_scripts/normalizePval_rescaled_WT_092020.png)

#Compare p-values for different features
These plots are exploratory checks to see if the p-values for different features are correlated with one another. I didn't find anything too interesting here.
```{r plotCompareScores, eval=F}
plotCompareScores(LUHMES_annotation_rescaled, x_name="expression", y_name="methylation_promoter")
plotCompareScores(LUHMES_annotation_rescaled, x_name="expression", y_name="methylation_body")
plotCompareScores(LUHMES_annotation_rescaled, x_name="expression", y_name="methylation_enhancer")
plotCompareScores(LUHMES_annotation_rescaled, x_name="expression", y_name="hydroxy_body")
```

#Weight the features and re-score
Weights are specified a priori according to experimental context and research questions. Here, promoter/enhancer methylation and expression will be weighted higher, and body methylation will be weighted lower.
```{r weighting, eval=F}
LUHMES_annotation_scored_bidirectional <- scorePval(LUHMES_annotation_rescaled, weights = c(methylation_promoter=0.125, methylation_body=0.1, methylation_enhancer=0.125 , expression=0.4, hydroxy_promoter=0.075, hydroxy_body=0.1, hydroxy_enhancer=0.075))
save(LUHMES_annotation_scored_bidirectional, file="~/KoborLab/kobor_space/sschaffner/LUHMES_201806/clean_scripts/LUHMES_annotation_scored_bidirectional_WT_092020.RData")
```

#Visualize modules
NOTE: The runSpinglass function has caused my RStudio session on the koborlab-hpc servers to freeze in the past. I recommend running that function in Terminal where it won't crash, then loading the object back into RStudio for plotting.
```{r visualize modules, eval=F}
#######To run in Terminal
library(SMITE)
load("~/KoborLab/kobor_space/sschaffner/LUHMES_201806/clean_scripts/LUHMES_annotation_scored_bidirectional_WT_092020.RData")

#load REACTOME interactome
load(system.file("data", "Reactome.Symbol.Igraph.rda", package = "SMITE"))

#Run spin-glass algorithm to determine best modules
LUHMES_spinglass <- runSpinglass(LUHMES_annotation_scored_bidirectional, network = REACTOME, maxsize = 50, num_iterations = 1000, random_alpha=0.05, node_alpha=0.05)
save(LUHMES_spinglass, file="~/KoborLab/kobor_space/sschaffner/LUHMES_201806/clean_scripts/LUHMES_spinglass_WT_092020.RData")
###########################

load("~/KoborLab/kobor_space/sschaffner/LUHMES_201806/clean_scripts/LUHMES_spinglass_WT_092020.RData")

#Run GOseq on individual modules
LUHMES_spinglass <- runGOseq(LUHMES_spinglass, coverage = read.table(system.file("extdata", "hg19_symbol_hpaii.sites.inbodyand2kbupstream.bed.gz", package = "SMITE")), type = "reactome")
GO_results <- extractGOseq(LUHMES_spinglass)

#Explore result names or search for a specific GO term using code below
names(GO_results) #Lists interaction modules
# [1] "NOTCH2:NOTCH3"                           
# [2] "CCND1:CKS1B:E2F1"                        
# [3] "H2AFX:BRCA1:BRCA2:RAD51"                 
# [4] "KIF4A:KIF11:KIF23:KLC1:KIF4B:KIFC1:KIF15"
# [5] "DLL1:DLL4"                               
# [6] "YAP1:STK3:WWTR1"                         
# [7] "ACTN3"                                   
# [8] "CTNNA2:CDH15:BOC"                        
# [9] "DLG3"                                    
#[10] "GRIK2"                                   
#[11] "MYL1:MYH3:MYL4"                          
#[12] "USP1:FANCB:FANCD2:FANCE:FANCI"           
#[13] "SMC2:NCAPD3:NCAPH:NCAPD2:NCAPG:SMC4"     
#[14] "IRS1"                                    
#[15] "IGF2:IGFBP5"                             
#[16] "PDGFRB:PDGFRA"                           
#[17] "TP53:MRE11A"                             
#[18] "P4HB"      

save(LUHMES_spinglass, file="~/KoborLab/kobor_space/sschaffner/LUHMES_201806/clean_scripts/LUHMES_spinglass_WT_092020.RData")
```

##Plotting modules
Change the "which_network" argument to plot each of the different networks generated.

Below, a smaller plot of the GRIK2 module is shown (to be used in a figure profiling this gene). Find images of all modules in the folder "~/KoborLab/kobor_space/sschaffner/LUHMES_201806/clean_scripts/SNCA_OE_modules" (all modules will be included in Supplementary material).
```{r plotmodule, eval=F}
plotModule2(LUHMES_spinglass, which_network=10, layout = "circle", label_scale = TRUE, compare_plot = FALSE, suppress_details = TRUE)
```
![WT module 10 - GRIK2](/home/BCRICWH.LAN/sschaffner/KoborLab/kobor_space/sschaffner/LUHMES_201806/clean_scripts/WT_module10_small.png)

##Extracting detailed module information
```{r}
load("~/KoborLab/kobor_space/sschaffner/LUHMES_201806/clean_scripts/LUHMES_spinglass_WT_092020.RData")
pval_df <- LUHMES_spinglass@score_data@pval_data
genes <- LUHMES_spinglass@score_data@genes
rownames(pval_df) <- genes

effect_df <- LUHMES_spinglass@score_data@effect_data
rownames(effect_df) <- genes

#getting only genes from significant modules
modules <- LUHMES_spinglass@score_data@module_output$modules
sig_genes <- unique(unlist(modules)) #351 genes
pval_df_sub <- pval_df[rownames(pval_df) %in% sig_genes,]
effect_df_sub <- effect_df[rownames(effect_df) %in% sig_genes,]

#annotating modules
pval_df_sub$gene <- rownames(pval_df_sub)
pval_df_sub$module <- NA
pval_df_sub$module <- sapply(1:nrow(pval_df_sub), function(x) names(modules[grep(pval_df_sub$gene[x],modules)]))
#some genes belong to multiple modules - fix this
pval_df_sub$num_modules <- sapply(1:nrow(pval_df_sub), function(x) length(unlist(pval_df_sub$module[x])))
nrow(pval_df_sub[pval_df_sub$num_modules>1,]) #119 genes belong to multiple modules

#plot p-values
library(reshape2)
library(ggplot2)
pval_df_melt <- melt(pval_df_sub[,1:8], id.vars="gene")
pval_df_melt$mod <- NA
pval_df_melt[grep("methylation", pval_df_melt$variable),"mod"] <- "DNAm"
pval_df_melt[grep("hydroxy", pval_df_melt$variable),"mod"] <- "DNAhm"
pval_df_melt[grep("expression", pval_df_melt$variable),"mod"] <- "mRNA"

ggplot(pval_df_melt, aes(x=value, fill=mod, col=mod)) + geom_density(show.legend = FALSE) + facet_wrap(~variable) + theme_bw() + ggtitle("WT aSyn SMITE Modules")

promoter <- ggplot(pval_df_melt[grep("promoter",pval_df_melt$variable),], aes(x=value, fill=mod, col=mod)) + geom_density(show.legend = FALSE, alpha=0.2) + theme_bw() + ggtitle("WT aSyn SMITE Modules")

#categorical variable summarizing sig/non-sig categories
pval_df_summary <- pval_df_sub[,1:8]

pval_df_summary$DNAm_promoter <- ">0.05"
pval_df_summary[pval_df_summary$methylation_promoter_pvalue<=0.05 & complete.cases(pval_df_summary$methylation_promoter_pvalue),"DNAm_promoter"] <- "<=0.05"
pval_df_summary$DNAm_body <- ">0.05"
pval_df_summary[pval_df_summary$methylation_body_pvalue<=0.05 & complete.cases(pval_df_summary$methylation_body_pvalue),"DNAm_body"] <- "<=0.05"
pval_df_summary$DNAm_enhancer <- ">0.05"
pval_df_summary[pval_df_summary$methylation_enhancer_pvalue<=0.05 & complete.cases(pval_df_summary$methylation_enhancer_pvalue),"DNAm_enhancer"] <- "<=0.05"

pval_df_summary$DNAhm_promoter <- ">0.05"
pval_df_summary[pval_df_summary$hydroxy_promoter_pvalue<=0.05 & complete.cases(pval_df_summary$hydroxy_promoter_pvalue),"DNAhm_promoter"] <- "<=0.05"
pval_df_summary$DNAhm_body <- ">0.05"
pval_df_summary[pval_df_summary$hydroxy_body_pvalue<=0.05 & complete.cases(pval_df_summary$hydroxy_body_pvalue),"DNAhm_body"] <- "<=0.05"
pval_df_summary$DNAhm_enhancer <- ">0.05"
pval_df_summary[pval_df_summary$hydroxy_enhancer_pvalue<=0.05 & complete.cases(pval_df_summary$hydroxy_enhancer_pvalue),"DNAhm_enhancer"] <- "<=0.05"

pval_df_summary$Expression <- ">0.05"
pval_df_summary[pval_df_summary$expression_pvalue<=0.05 & complete.cases(pval_df_summary$expression_pvalue),"Expression"] <- "<=0.05"

unlist(apply(pval_df_summary[,9:15], 2, function(x) summary(as.factor(x))))
#DNAm prom: 6 genes
#DNAm body: 44 genes
#DNAm enhancer: 10 genes
#DNAhm promoter: 0 genes
#DNAhm body: 2 genes
#DNAhm enhancer: 1 gene
#Expression: 94 genes


sig_genes <- unique(unlist(apply(pval_df_summary[,9:15], 2, function(x) grep("<=0.05",x))))
nrow(pval_df_summary_sig <- pval_df_summary[sig_genes,]) #136

pval_df_melt <- melt(pval_df_summary_sig[,8:15], id.vars="gene")
ggplot(pval_df_melt, aes(x=value, fill=value)) + geom_bar(stat="count", show.legend=FALSE, col="black") + facet_wrap(~variable) + theme_classic() + ylab("Number of Genes") + xlab("P-value") + scale_fill_manual(values=c("grey","white")) + geom_text(stat="count", aes(label = after_stat(count)), vjust=-0.5, size=3.5) + ylim(c(0,160)) + ggtitle("Significant WT aSyn Module Genes")

###Significant DNAm changes
nrow(pval_df_sub[pval_df_sub$methylation_promoter_pvalue<=0.05,]) #14 genes
nrow(pval_df_sub[pval_df_sub$methylation_body_pvalue<=0.05,]) #62
nrow(pval_df_sub[pval_df_sub$methylation_enhancer_pvalue<=0.05,]) #212 genes

###Significant DNAhm changes
nrow(pval_df_sub[pval_df_sub$hydroxy_promoter_pvalue<=0.05,]) #156 genes
nrow(pval_df_sub[pval_df_sub$hydroxy_body_pvalue<=0.05,]) #54
nrow(pval_df_sub[pval_df_sub$hydroxy_enhancer_pvalue<=0.05,]) #245 genes

###Significant expression changes
nrow(pval_df_sub[pval_df_sub$expression_pvalue<=0.05,]) #144 genes

###Significant expression and DNAm changes
expr_DNAm_prom <- pval_df_sub[pval_df_sub$expression_pvalue<=0.05 & pval_df_sub$methylation_promoter_pvalue<=0.05,]
unique(expr_DNAm_prom$gene[complete.cases(expr_DNAm_prom$gene)])
#2 genes: GRIK2, IGF2

expr_DNAm_body <- pval_df_sub[pval_df_sub$expression_pvalue<=0.05 & pval_df_sub$methylation_body_pvalue<=0.05,]
unique(expr_DNAm_body$gene[complete.cases(expr_DNAm_body$gene)])
#13 genes
# [1] "BOC"    "CDH3"   "CDK6"   "CTNNA2" "DLG3"   "DNER"   "DTX1"   "GRIK2"  "HDAC9" 
#[10] "IRS1"   "MAGI2"  "PDGFRA" "WWTR1" 

expr_DNAm_enh <- pval_df_sub[pval_df_sub$expression_pvalue<=0.05 & pval_df_sub$methylation_enhancer_pvalue<=0.05,]
unique(expr_DNAm_enh$gene[complete.cases(expr_DNAm_enh$gene)]) #"IGF2"

DNAm_genes <- unique(c(expr_DNAm_prom$gene, expr_DNAm_body$gene, expr_DNAhm_enh$gene))
DNAm_genes[complete.cases(DNAm_genes)] #15 genes
# [1] "GRIK2"  "IGF2"   "BOC"    "CDH3"   "CDK6"   "CTNNA2" "DLG3"   "DNER"   "DTX1"  
#[10] "HDAC9"  "IRS1"   "MAGI2"  "PDGFRA" "WWTR1"  "GPS2"  

#pvalue histograms
ggplot(pval_df_melt[pval_df_melt$gene %in% DNAm_genes,], aes(x=value, fill=mod)) + geom_histogram(show.legend = FALSE) + facet_wrap(~variable) + theme_bw() + ggtitle("WT aSyn SMITE Modules with Changes in DNAm and Expression")

#scatterplots with expression pvalues
pval_df_melt2 <- melt(pval_df_sub[,1:8], id.vars=c("gene","expression_pvalue"))
pval_df_melt2$mod <- NA
pval_df_melt2[grep("methylation", pval_df_melt2$variable),"mod"] <- "DNAm"
pval_df_melt2[grep("hydroxy", pval_df_melt2$variable),"mod"] <- "DNAhm"

ggplot(pval_df_melt2[pval_df_melt2$gene %in% DNAm_genes,], aes(x=value, y=expression_pvalue, col=mod)) + geom_point(show.legend = FALSE) + facet_wrap(~variable) + theme_bw() + ggtitle("WT aSyn SMITE Modules with Changes in DNAm and Expression")

label_data <- pval_df_melt2[pval_df_melt2$expression_pvalue<=0.05 & pval_df_melt2$value<=0.05,]
label_data <- label_data[complete.cases(label_data$variable),]

library(ggrepel) #for labelling text
#library(ggshadow) #for glowing points
ggplot(pval_df_melt2[complete.cases(pval_df_melt2$gene),], aes(x=value, y=expression_pvalue, col=mod)) + geom_hline(yintercept=0, color="grey") + geom_vline(xintercept=0, color="grey") + geom_hline(yintercept=0.05, color="grey", linetype="dashed") + geom_vline(xintercept=0.05, color="grey", linetype="dashed") +  geom_point(show.legend = FALSE) + facet_wrap(~variable) + theme_bw() + ggrepel::geom_text_repel(data=label_data, label=label_data$gene, col="black", size=3, max.overlaps = Inf) + coord_cartesian(clip = "off") + scale_x_continuous(breaks=c(0,0.2,0.4,0.6,0.8,1), limits=c(-0.2,1)) + scale_y_continuous(breaks=c(0,0.2,0.4,0.6,0.8,1), limits=c(-0.2,1)) + ggtitle("WT aSyn SMITE Modules") + theme(panel.grid.minor.x = element_blank(), panel.grid.minor.y=element_blank())

#scatterplots with expression effects
summary(effect_df_sub$expression_effect) #-1.77 to 1.08
summary(effect_df_sub$methylation_promoter_effect) #-0.09 to 0.11
summary(effect_df_sub$hydroxy_promoter_effect) #-0.03 to 0.04

effect_df_sub$gene <- rownames(effect_df_sub)
effect_df_melt2 <- melt(effect_df_sub[,1:8], id.vars=c("gene","expression_effect"))
effect_df_melt2$mod <- NA
effect_df_melt2[grep("methylation", effect_df_melt2$variable),"mod"] <- "DNAm"
effect_df_melt2[grep("hydroxy", effect_df_melt2$variable),"mod"] <- "DNAhm"

ggplot(effect_df_melt2[effect_df_melt2$gene %in% DNAm_genes,], aes(x=value, y=expression_pvalue, col=mod)) + geom_point(show.legend = FALSE) + facet_wrap(~variable) + theme_bw() + ggtitle("WT aSyn SMITE Modules with Changes in DNAm and Expression")

#label_data <- effect_df_melt2[abs(effect_df_melt2$expression_pvalue<=0.05 & effect_df_melt2$value<=0.05,]
#label_data <- label_data[complete.cases(label_data$variable),]

library(ggrepel) #for labelling text
#library(ggshadow) #for glowing points
label_data$gene_var <- paste(label_data$gene, label_data$variable)
label_data_effect <- effect_df_melt2[effect_df_melt2$gene %in% label_data$gene,]
label_data_effect$gene_var <- paste(label_data_effect$gene, label_data_effect$variable)
label_data_effect$gene_var <- gsub("effect","pvalue",label_data_effect$gene_var)
label_data_effect <- label_data_effect[label_data_effect$gene_var %in% label_data$gene_var,]

ggplot(effect_df_melt2[complete.cases(effect_df_melt2$gene),], aes(x=value, y=expression_effect, col=mod)) + geom_hline(yintercept=0, color="grey") + geom_vline(xintercept=0, color="grey") +  geom_point(show.legend = FALSE) + facet_wrap(~variable) + theme_bw() + ggrepel::geom_text_repel(data=label_data_effect, label=label_data_effect$gene, col="black", size=3, max.overlaps = Inf) + coord_cartesian(clip = "off") + ggtitle("WT aSyn SMITE Modules")

#no labels
#label_data <- effect_df_melt2[abs(effect_df_melt2$expression_effect)>=0.5,]
#label_data <- label_data[complete.cases(label_data$variable),]


ggplot(effect_df_melt2[complete.cases(effect_df_melt2$gene),], aes(x=value, y=expression_effect, col=mod)) + ggtitle("WT aSyn SMITE Modules") + theme(panel.grid.minor.x = element_blank(), panel.grid.minor.y=element_blank()) +  geom_point(show.legend = FALSE) + facet_wrap(~variable) + theme_bw() + geom_hline(yintercept=0, color="black") + geom_vline(xintercept=0, color="black")

###Significant expression and DNAhm changes
nrow(expr_DNAhm_prom <- pval_df_sub[pval_df_sub$expression_pvalue<=0.05 & pval_df_sub$hydroxy_promoter_pvalue<=0.05,]) #69 genes
nrow(expr_DNAhm_body <- pval_df_sub[pval_df_sub$expression_pvalue<=0.05 & pval_df_sub$hydroxy_body_pvalue<=0.05,]) #33 genes
nrow(expr_DNAhm_enh <- pval_df_sub[pval_df_sub$expression_pvalue<=0.05 & pval_df_sub$hydroxy_enhancer_pvalue<=0.05,]) #106 genes

pval_df_sub[grep("GRIK2", pval_df_sub$gene),]
#      methylation_promoter_pvalue methylation_body_pvalue methylation_enhancer_pvalue
#GRIK2                        0.01                    0.01                          NA
#      hydroxy_promoter_pvalue hydroxy_body_pvalue hydroxy_enhancer_pvalue expression_pvalue  gene
#GRIK2               0.9791056          0.09757986                      NA              0.01 GRIK2
#           module num_modules
#GRIK2 DLG3, GRIK2           2

#DNAm and expression changes were scored as "significant" by SMITE; hydroxy was not, although there was a significant site-specific change in DNAhm when EPIC data was investigated. There was no enhancer around GRIK2.

###Plotting DNA

ggplot(pval_df_melt, aes(x=value, fill=mod)) + geom_histogram(show.legend = FALSE) + facet_wrap(~variable) + theme_bw() + ggtitle("WT aSyn SMITE Modules")
```
![Summary of significant SMITE module genes by 'omic](/home/BCRICWH.LAN/sschaffner/KoborLab/kobor_space/sschaffner/LUHMES_201806/clean_scripts/module_pvals_WT_barplot.png)

![P-values of DNAm, DNAhm, and expression in genes from significant SMITE modules](/home/BCRICWH.LAN/sschaffner/KoborLab/kobor_space/sschaffner/LUHMES_201806/clean_scripts/module_pvals_WT_scatter_label.png)

![Effect sizes of DNAm, DNAhm, and expression in genes from significant SMITE modules](/home/BCRICWH.LAN/sschaffner/KoborLab/kobor_space/sschaffner/LUHMES_201806/clean_scripts/module_effects_WT_scatter_label.png)

##Locations of enhancers
```{r}
anno <- as.data.frame(LUHMES_spinglass@annotation@unlistData)
anno_enh <- anno[anno$feature=="enhancer",]
anno_enh[anno_enh$name=="GRIK2",] #nothing annotated to GRIK2
#check by coordinates from GRIK2 plot
#chr6: 101846073-102516590
anno_enh[anno_enh$seqnames=="chr6" & anno_enh$start>=(101846073-5000) & anno_enh$end<=(102516590+5000),] #nothing, no peaks here.

#checking H3K4me1 consensus peaks in all genotypes
enh_GRIK2 <- enhancer[enhancer$seqnames=="chr6" & enhancer$start>=(101846073-5000) & enhancer$end<=(102516590+5000),]
#       seqnames     start       end width strand        GFP        WT       A30P
#163112     chr6 101979580 101980052   473      * 0.00000000 0.0000000 0.07521559
#163113     chr6 102390019 102390755   737      * 0.06173119 0.0000000 0.07639720
#163114     chr6 102458334 102459567  1234      * 0.06845595 0.1383643 0.06361120

#GFP peak areas in SMITE annotation
anno[anno$seqnames=="chr6" & anno$start>=102390019 & anno$end<=102390755,] #no rows
anno[anno$seqnames=="chr6" & anno$start>=102458334 & anno$end<=102459567,] #no rows

#GFP peak areas in EPIC fdat
library(methylumi)
load("~/KoborLab/kobor_space/sschaffner/LUHMES_201806/clean_scripts/LUHMES_mC_clean_SS_06182020.RData")
fdat <- fData(mC_methylumi_N)
fdat[fdat$CHR==6 & fdat$MAPINFO>=102390019 & fdat$MAPINFO<=102390755,] #no rows
fdat[fdat$CHR==6 & fdat$MAPINFO>=102458334 & fdat$MAPINFO<=102459567,] #no rows

####Plotting consensus peak scores near GRIK2
GFP <- ggplot(enh_GRIK2[,1:6]) + geom_rect(aes(xmin=start, xmax=end, ymin=0, ymax=1, fill=GFP)) + theme_classic() + theme(axis.text.y=element_blank(), axis.ticks.y = element_blank()) + xlab("chr6 coordinate") + scale_fill_distiller(direction=1, name="Score", limit=c(0,0.1383643)) + ylab("Control")
WT <- ggplot(enh_GRIK2[,c(1:5,7)]) + geom_rect(aes(xmin=start, xmax=end, ymin=0, ymax=1, fill=WT)) + theme_classic() + theme(axis.text.y=element_blank(), axis.ticks.y=element_blank()) + xlab("chr6 coordinate") + scale_fill_distiller(direction=1, name="Score", limit=c(0,0.1383643)) + ylab("WT aSyn")
A30P <- ggplot(enh_GRIK2[,c(1:5,8)]) + geom_rect(aes(xmin=start, xmax=end, ymin=0, ymax=1, fill=A30P)) + theme_classic() + theme(axis.text.y=element_blank(), axis.ticks.y=element_blank()) + xlab("chr6 coordinate") + scale_fill_distiller(direction=1, name="Score", limit=c(0,0.1383643)) + ylab("A30P aSyn")
grid.arrange(GFP,WT,A30P,nrow=3)
```
There are three H3K4me1 consensus peaks in the GRIK2 gene; GFP cells contain two peaks, WT cells contain one peak, and A30P cells contain all three. The two GFP peaks are not represented in the SMITE annotation. This is because there are no EPIC array probes within 5kb of these peaks. Keep in mind that SMITE results are restricted to DNAm changes detectable on the array and where these are occurring!

![H3K4me1 consensus peaks near GRIK2](/home/BCRICWH.LAN/sschaffner/KoborLab/kobor_space/sschaffner/LUHMES_201806/clean_scripts/GRIK2_ChIP.png)
