DecipherPD LUHMES EPIC Array Runs - Annotation of longest transcript per gene
================================
  
##### Analyst: Samantha Schaffner
##### Date: June 18, 2020
  
This analysis contains EPIC BeadChip array data on Lund human mesencephalic (LUHMES) cells, differentiated into dopamnergic neurons, for the DecipherPD consortium. Tiago Outerio's group (Gottingen, Germany) has created LUHMES overexpressing wild type or A30P mutant human alpha-synuclein (a-Syn) as well as GFP. This project compares control+GFP versus WTa-Syn+GFP cells or A30Pa-Syn+GFP cells; the data consists of seven biological replicates of control cells, eight biological replicates each of SNCA-OE and A30P cells, and one technical replicate of SNCA-OE. Each of these samples were run as bisulfite and oxidative bisulfite-converted aliquots (total = 24 samples*2 conversion types = 48 reactions).

In this script, I examined all of the sites analyzed for DNA methylation with the goal of collapsing the gene/genomic context annotation into one entry per site. I used the longest transcript of each gene to capture the most genomic information (see Edgar et al., Epigenetics & Chromatin 2014).

#Libraries and objects
```{r libraries, eval=F}
setwd("~/")
library(biomaRt)
load("~/KoborLab/kobor_space/sschaffner/LUHMES_201806/clean_scripts/transcript_df_062020.RData")
```

#Set up database and query
Illumina annotation should be for hg19, so I'll use the hg19 version of the biomaRt database.
```{r database, eval=F}
#accessing biomaRt database for hg19 genes - will reference this to obtain transcript length
mart <- useMart("ENSEMBL_MART_ENSEMBL", dataset="hsapiens_gene_ensembl", host="feb2014.archive.ensembl.org",path="/biomart/martservice",archive=FALSE, verbose=TRUE)
#Filters: UCSC refgene accession (want a specific transcript)
#Attributes: refgene accession, transcript length

#getting all the sites that have multiple transcripts annotated to them
length(unique_cpgs <- unique(transcript_df$site[which(duplicated(transcript_df$site))])) #297,859

#subsetting the master annotation to sites with multiple annotations
nrow(transcript_df_sub <- transcript_df[transcript_df$site %in% unique_cpgs,]) #1,007,243

#obtaining a list of all the cpgs which need to be annotated by transcript
unique_cpgs <- data.frame(site=unique_cpgs)
```

#Transcript length calculation: Protein-coding genes
In order to calculate the longest transcript per CpG site, the following steps were performed:
1. Subset the unique CpG sites into groups of 10,000 at a time for annotation. This will ensure that the query to the biomaRt database does not crash, and the R session does not crash from building a large data frame inside a loop. 
2. Looping over the CpG sites, obtain all of the transcript annotations for one site at a time, then use biomaRt to calculate their lengths and determine the longest one.
3. Make a vector of the longest transcripts per site (merging the data from each loop), and put that together with the list of CpG sites queried, to create a dataframe.

Note: In this section, I queried the mRNA database of biomaRt. Some sites are located in noncoding RNA genes and won't have an annotation here; I will annotate those in the next section.
```{r length calculation coding, eval=F}
longest_transcript <- sapply(1:10000, function(i) {
  
  #only put a print statement into the first couple of loops to see how long it was taking...commented out later on to increase efficiency
  #print(i)
  
  #getting all transcript annotations for a cpg site
  cpg_set <- transcript_df[transcript_df$site==unique_cpgs$site[i],]
  
  #querying biomaRt database to get the length of each transcript
  #cpg_set_length <- getBM(mart=human_mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_length"))
  
  cpg_set_length <- getBM(mart=mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_start", "transcript_end"))
  cpg_set_length$transcript_length = cpg_set_length$transcript_end - cpg_set_length$transcript_start
  
  #if multiple transcripts, find the maximum transcript length
  if (nrow(cpg_set_length)>1){ max_length_acc = cpg_set_length[cpg_set_length$transcript_length==max(cpg_set_length$transcript_length),"refseq_mrna"]
  
  #if there is one unique longest transcript, return that one
  if (length(max_length_acc)==1) { return(max_length_acc) }
  
  #if there are duplicate transcripts in the database with the maximum length, return one of the two annotations
  else { return(max_length_acc[1]) }}
  
  #if there is only one transcript per cpg site, return that transcript
  else if (nrow(cpg_set_length)==1){ return(cpg_set_length$refseq_mrna) }
  
  #if there are no transcripts associated with a cpg, return "none"
  else { return("none") }
  #else { stop() }
})

#double checking 2 things: 1) the length of the "longest transcript" vector, AKA ensuring we have a "longest transcript" value for every CpG site that was queried; 2) checking that there is one entry per CpG site (no missing or duplicated values)
lengths <- lapply(1:length(longest_transcript), function(x) length(unlist(longest_transcript[x])))
lengths <- unlist(lengths)
summary(as.factor(lengths))
#there is one value for every site!

longest_transcript <- unlist(longest_transcript)

longest_transcript2 <- sapply(10001:20000, function(i) {
  
  #print(i)
  cpg_set <- transcript_df[transcript_df$site==unique_cpgs$site[i],]
  #cpg_set_length <- getBM(mart=human_mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_length"))
  
  cpg_set_length <- getBM(mart=mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_start", "transcript_end"))
  cpg_set_length$transcript_length = cpg_set_length$transcript_end - cpg_set_length$transcript_start
  
  if (nrow(cpg_set_length)>1){ max_length_acc = cpg_set_length[cpg_set_length$transcript_length==max(cpg_set_length$transcript_length),"refseq_mrna"]
  if (length(max_length_acc)==1) { return(max_length_acc) }
  else { return(max_length_acc[1]) }}
  
  else if (nrow(cpg_set_length)==1){ return(cpg_set_length$refseq_mrna) }
  else { return("none") }
  #else { stop() }
})
longest_transcript2 <- unlist(longest_transcript2)
longest_transcript <- c(longest_transcript, longest_transcript2)

longest_transcript3 <- sapply(20001:30000, function(i) {
  
  #print(i)
  cpg_set <- transcript_df[transcript_df$site==unique_cpgs$site[i],]
  #cpg_set_length <- getBM(mart=human_mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_length"))
  
  cpg_set_length <- getBM(mart=mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_start", "transcript_end"))
  cpg_set_length$transcript_length = cpg_set_length$transcript_end - cpg_set_length$transcript_start
  
  if (nrow(cpg_set_length)>1){ max_length_acc = cpg_set_length[cpg_set_length$transcript_length==max(cpg_set_length$transcript_length),"refseq_mrna"]
  if (length(max_length_acc)==1) { return(max_length_acc) }
  else { return(max_length_acc[1]) }}
  
  else if (nrow(cpg_set_length)==1){ return(cpg_set_length$refseq_mrna) }
  else { return("none") }
  #else { stop() }
})
longest_transcript3 <- unlist(longest_transcript3)
longest_transcript <- c(longest_transcript, longest_transcript3)

longest_transcript4 <- sapply(30001:40000, function(i) {
  
  #print(i)
  cpg_set <- transcript_df[transcript_df$site==unique_cpgs$site[i],]
  #cpg_set_length <- getBM(mart=human_mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_length"))
  
  cpg_set_length <- getBM(mart=mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_start", "transcript_end"))
  cpg_set_length$transcript_length = cpg_set_length$transcript_end - cpg_set_length$transcript_start
  
  if (nrow(cpg_set_length)>1){ max_length_acc = cpg_set_length[cpg_set_length$transcript_length==max(cpg_set_length$transcript_length),"refseq_mrna"]
  if (length(max_length_acc)==1) { return(max_length_acc) }
  else { return(max_length_acc[1]) }}
  
  else if (nrow(cpg_set_length)==1){ return(cpg_set_length$refseq_mrna) }
  else { return("none") }
  #else { stop() }
})
longest_transcript4 <- unlist(longest_transcript4)
longest_transcript <- c(longest_transcript, longest_transcript4)

longest_transcript5 <- sapply(40001:50000, function(i) {
  
  #print(i)
  cpg_set <- transcript_df[transcript_df$site==unique_cpgs$site[i],]
  #cpg_set_length <- getBM(mart=human_mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_length"))
  
  cpg_set_length <- getBM(mart=mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_start", "transcript_end"))
  cpg_set_length$transcript_length = cpg_set_length$transcript_end - cpg_set_length$transcript_start
  
  if (nrow(cpg_set_length)>1){ max_length_acc = cpg_set_length[cpg_set_length$transcript_length==max(cpg_set_length$transcript_length),"refseq_mrna"]
  if (length(max_length_acc)==1) { return(max_length_acc) }
  else { return(max_length_acc[1]) }}
  
  else if (nrow(cpg_set_length)==1){ return(cpg_set_length$refseq_mrna) }
  else { return("none") }
  #else { stop() }
})
longest_transcript5 <- unlist(longest_transcript5)
longest_transcript <- c(longest_transcript, longest_transcript5)
rm(longest_transcript2, longest_transcript3, longest_transcript4, longest_transcript5)

longest_transcript6 <- sapply(50001:60000, function(i) {
  
  #print(i)
  cpg_set <- transcript_df[transcript_df$site==unique_cpgs$site[i],]
  #cpg_set_length <- getBM(mart=human_mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_length"))
  
  cpg_set_length <- getBM(mart=mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_start", "transcript_end"))
  cpg_set_length$transcript_length = cpg_set_length$transcript_end - cpg_set_length$transcript_start
  
  if (nrow(cpg_set_length)>1){ max_length_acc = cpg_set_length[cpg_set_length$transcript_length==max(cpg_set_length$transcript_length),"refseq_mrna"]
  if (length(max_length_acc)==1) { return(max_length_acc) }
  else { return(max_length_acc[1]) }}
  
  else if (nrow(cpg_set_length)==1){ return(cpg_set_length$refseq_mrna) }
  else { return("none") }
  #else { stop() }
})
longest_transcript6 <- unlist(longest_transcript6)
longest_transcript <- c(longest_transcript, longest_transcript6)

longest_transcript7 <- sapply(60001:70000, function(i) {
  
  #print(i)
  cpg_set <- transcript_df[transcript_df$site==unique_cpgs$site[i],]
 # cpg_set_length <- getBM(mart=human_mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_length"))
  
  cpg_set_length <- getBM(mart=mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_start", "transcript_end"))
  cpg_set_length$transcript_length = cpg_set_length$transcript_end - cpg_set_length$transcript_start
  
  if (nrow(cpg_set_length)>1){ max_length_acc = cpg_set_length[cpg_set_length$transcript_length==max(cpg_set_length$transcript_length),"refseq_mrna"]
  if (length(max_length_acc)==1) { return(max_length_acc) }
  else { return(max_length_acc[1]) }}
  
  else if (nrow(cpg_set_length)==1){ return(cpg_set_length$refseq_mrna) }
  else { return("none") }
  #else { stop() }
})
longest_transcript7 <- unlist(longest_transcript7)
longest_transcript <- c(longest_transcript, longest_transcript7)

longest_transcript8 <- sapply(70001:80000, function(i) {
  
  #print(i)
  cpg_set <- transcript_df[transcript_df$site==unique_cpgs$site[i],]
  #cpg_set_length <- getBM(mart=human_mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_length"))
  
  cpg_set_length <- getBM(mart=mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_start", "transcript_end"))
  cpg_set_length$transcript_length = cpg_set_length$transcript_end - cpg_set_length$transcript_start
  
  if (nrow(cpg_set_length)>1){ max_length_acc = cpg_set_length[cpg_set_length$transcript_length==max(cpg_set_length$transcript_length),"refseq_mrna"]
  if (length(max_length_acc)==1) { return(max_length_acc) }
  else { return(max_length_acc[1]) }}
  
  else if (nrow(cpg_set_length)==1){ return(cpg_set_length$refseq_mrna) }
  else { return("none") }
  #else { stop() }
})
longest_transcript8 <- unlist(longest_transcript8)
longest_transcript <- c(longest_transcript, longest_transcript8)

longest_transcript9 <- sapply(80001:90000, function(i) {
  
  #print(i)
  cpg_set <- transcript_df[transcript_df$site==unique_cpgs$site[i],]
  #cpg_set_length <- getBM(mart=human_mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_length"))
  
  cpg_set_length <- getBM(mart=mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_start", "transcript_end"))
  cpg_set_length$transcript_length = cpg_set_length$transcript_end - cpg_set_length$transcript_start
  
  if (nrow(cpg_set_length)>1){ max_length_acc = cpg_set_length[cpg_set_length$transcript_length==max(cpg_set_length$transcript_length),"refseq_mrna"]
  if (length(max_length_acc)==1) { return(max_length_acc) }
  else { return(max_length_acc[1]) }}
  
  else if (nrow(cpg_set_length)==1){ return(cpg_set_length$refseq_mrna) }
  else { return("none") }
  #else { stop() }
})
longest_transcript9 <- unlist(longest_transcript9)
longest_transcript <- c(longest_transcript, longest_transcript9)

longest_transcript10 <- sapply(90001:100000, function(i) {
  
  #print(i)
  cpg_set <- transcript_df[transcript_df$site==unique_cpgs$site[i],]
  #cpg_set_length <- getBM(mart=human_mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_length"))
  
  cpg_set_length <- getBM(mart=mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_start", "transcript_end"))
  cpg_set_length$transcript_length = cpg_set_length$transcript_end - cpg_set_length$transcript_start
  
  if (nrow(cpg_set_length)>1){ max_length_acc = cpg_set_length[cpg_set_length$transcript_length==max(cpg_set_length$transcript_length),"refseq_mrna"]
  if (length(max_length_acc)==1) { return(max_length_acc) }
  else { return(max_length_acc[1]) }}
  
  else if (nrow(cpg_set_length)==1){ return(cpg_set_length$refseq_mrna) }
  else { return("none") }
  #else { stop() }
})
longest_transcript10 <- unlist(longest_transcript10)
longest_transcript <- c(longest_transcript, longest_transcript10)
rm(longest_transcript6, longest_transcript7, longest_transcript8, longest_transcript9, longest_transcript10)

######################################################################

longest_transcript_11 <- sapply(100001:110000, function(i) {
  
  #print(i)
  cpg_set <- transcript_df[transcript_df$site==unique_cpgs$site[i],]
  #cpg_set_length <- getBM(mart=human_mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_length"))
  
  cpg_set_length <- getBM(mart=mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_start", "transcript_end"))
  cpg_set_length$transcript_length = cpg_set_length$transcript_end - cpg_set_length$transcript_start
  
  if (nrow(cpg_set_length)>1){ max_length_acc = cpg_set_length[cpg_set_length$transcript_length==max(cpg_set_length$transcript_length),"refseq_mrna"]
  if (length(max_length_acc)==1) { return(max_length_acc) }
  else { return(max_length_acc[1]) }}
  
  else if (nrow(cpg_set_length)==1){ return(cpg_set_length$refseq_mrna) }
  else { return("none") }
  #else { stop() }
})

longest_transcript_11 <- unlist(longest_transcript_11)
longest_transcript <- c(longest_transcript, longest_transcript_11)

longest_transcript12 <- sapply(110001:120000, function(i) {
  
  #print(i)
  cpg_set <- transcript_df[transcript_df$site==unique_cpgs$site[i],]
  #cpg_set_length <- getBM(mart=human_mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_length"))
  
  cpg_set_length <- getBM(mart=mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_start", "transcript_end"))
  cpg_set_length$transcript_length = cpg_set_length$transcript_end - cpg_set_length$transcript_start
  
  if (nrow(cpg_set_length)>1){ max_length_acc = cpg_set_length[cpg_set_length$transcript_length==max(cpg_set_length$transcript_length),"refseq_mrna"]
  if (length(max_length_acc)==1) { return(max_length_acc) }
  else { return(max_length_acc[1]) }}
  
  else if (nrow(cpg_set_length)==1){ return(cpg_set_length$refseq_mrna) }
  else { return("none") }
  #else { stop() }
})
longest_transcript12 <- unlist(longest_transcript12)
longest_transcript <- c(longest_transcript, longest_transcript12)

longest_transcript13 <- sapply(120001:130000, function(i) {
  
  #print(i)
  cpg_set <- transcript_df[transcript_df$site==unique_cpgs$site[i],]
  #cpg_set_length <- getBM(mart=human_mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_length"))
  
  cpg_set_length <- getBM(mart=mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_start", "transcript_end"))
  cpg_set_length$transcript_length = cpg_set_length$transcript_end - cpg_set_length$transcript_start
  
  if (nrow(cpg_set_length)>1){ max_length_acc = cpg_set_length[cpg_set_length$transcript_length==max(cpg_set_length$transcript_length),"refseq_mrna"]
  if (length(max_length_acc)==1) { return(max_length_acc) }
  else { return(max_length_acc[1]) }}
  
  else if (nrow(cpg_set_length)==1){ return(cpg_set_length$refseq_mrna) }
  else { return("none") }
  #else { stop() }
})
longest_transcript13 <- unlist(longest_transcript13)
longest_transcript <- c(longest_transcript, longest_transcript13)

longest_transcript14 <- sapply(130001:140000, function(i) {
  
  #print(i)
  cpg_set <- transcript_df[transcript_df$site==unique_cpgs$site[i],]
  #cpg_set_length <- getBM(mart=human_mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_length"))
  
  cpg_set_length <- getBM(mart=mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_start", "transcript_end"))
  cpg_set_length$transcript_length = cpg_set_length$transcript_end - cpg_set_length$transcript_start
  
  if (nrow(cpg_set_length)>1){ max_length_acc = cpg_set_length[cpg_set_length$transcript_length==max(cpg_set_length$transcript_length),"refseq_mrna"]
  if (length(max_length_acc)==1) { return(max_length_acc) }
  else { return(max_length_acc[1]) }}
  
  else if (nrow(cpg_set_length)==1){ return(cpg_set_length$refseq_mrna) }
  else { return("none") }
  #else { stop() }
})
longest_transcript14 <- unlist(longest_transcript14)
longest_transcript <- c(longest_transcript, longest_transcript14)

longest_transcript15 <- sapply(140001:150000, function(i) {
  
  #print(i)
  cpg_set <- transcript_df[transcript_df$site==unique_cpgs$site[i],]
  #cpg_set_length <- getBM(mart=human_mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_length"))
  
  cpg_set_length <- getBM(mart=mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_start", "transcript_end"))
  cpg_set_length$transcript_length = cpg_set_length$transcript_end - cpg_set_length$transcript_start
  
  if (nrow(cpg_set_length)>1){ max_length_acc = cpg_set_length[cpg_set_length$transcript_length==max(cpg_set_length$transcript_length),"refseq_mrna"]
  if (length(max_length_acc)==1) { return(max_length_acc) }
  else { return(max_length_acc[1]) }}
  
  else if (nrow(cpg_set_length)==1){ return(cpg_set_length$refseq_mrna) }
  else { return("none") }
  #else { stop() }
})
longest_transcript15 <- unlist(longest_transcript15)
longest_transcript <- c(longest_transcript, longest_transcript15)
rm(longest_transcript11, longest_transcript12, longest_transcript13, longest_transcript14, longest_transcript15)

longest_transcript16 <- sapply(150001:160000, function(i) {
  
  #print(i)
  cpg_set <- transcript_df[transcript_df$site==unique_cpgs$site[i],]
  #cpg_set_length <- getBM(mart=human_mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_length"))
  
  cpg_set_length <- getBM(mart=mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_start", "transcript_end"))
  cpg_set_length$transcript_length = cpg_set_length$transcript_end - cpg_set_length$transcript_start
  
  if (nrow(cpg_set_length)>1){ max_length_acc = cpg_set_length[cpg_set_length$transcript_length==max(cpg_set_length$transcript_length),"refseq_mrna"]
  if (length(max_length_acc)==1) { return(max_length_acc) }
  else { return(max_length_acc[1]) }}
  
  else if (nrow(cpg_set_length)==1){ return(cpg_set_length$refseq_mrna) }
  else { return("none") }
  #else { stop() }
})
longest_transcript16 <- unlist(longest_transcript16)
longest_transcript <- c(longest_transcript, longest_transcript16)

longest_transcript17 <- sapply(160001:170000, function(i) {
  
  #print(i)
  cpg_set <- transcript_df[transcript_df$site==unique_cpgs$site[i],]
  #cpg_set_length <- getBM(mart=human_mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_length"))
  
  cpg_set_length <- getBM(mart=mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_start", "transcript_end"))
  cpg_set_length$transcript_length = cpg_set_length$transcript_end - cpg_set_length$transcript_start
  
  if (nrow(cpg_set_length)>1){ max_length_acc = cpg_set_length[cpg_set_length$transcript_length==max(cpg_set_length$transcript_length),"refseq_mrna"]
  if (length(max_length_acc)==1) { return(max_length_acc) }
  else { return(max_length_acc[1]) }}
  
  else if (nrow(cpg_set_length)==1){ return(cpg_set_length$refseq_mrna) }
  else { return("none") }
  #else { stop() }
})
longest_transcript17 <- unlist(longest_transcript17)
longest_transcript <- c(longest_transcript, longest_transcript17)

longest_transcript18 <- sapply(170001:180000, function(i) {
  
  #print(i)
  cpg_set <- transcript_df[transcript_df$site==unique_cpgs$site[i],]
  #cpg_set_length <- getBM(mart=human_mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_length"))
  
  cpg_set_length <- getBM(mart=mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_start", "transcript_end"))
  cpg_set_length$transcript_length = cpg_set_length$transcript_end - cpg_set_length$transcript_start
  
  if (nrow(cpg_set_length)>1){ max_length_acc = cpg_set_length[cpg_set_length$transcript_length==max(cpg_set_length$transcript_length),"refseq_mrna"]
  if (length(max_length_acc)==1) { return(max_length_acc) }
  else { return(max_length_acc[1]) }}
  
  else if (nrow(cpg_set_length)==1){ return(cpg_set_length$refseq_mrna) }
  else { return("none") }
  #else { stop() }
})
longest_transcript18 <- unlist(longest_transcript18)
longest_transcript <- c(longest_transcript, longest_transcript18)

longest_transcript19 <- sapply(180001:190000, function(i) {
  
  #print(i)
  cpg_set <- transcript_df[transcript_df$site==unique_cpgs$site[i],]
  #cpg_set_length <- getBM(mart=human_mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_length"))
  
  cpg_set_length <- getBM(mart=mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_start", "transcript_end"))
  cpg_set_length$transcript_length = cpg_set_length$transcript_end - cpg_set_length$transcript_start
  
  if (nrow(cpg_set_length)>1){ max_length_acc = cpg_set_length[cpg_set_length$transcript_length==max(cpg_set_length$transcript_length),"refseq_mrna"]
  if (length(max_length_acc)==1) { return(max_length_acc) }
  else { return(max_length_acc[1]) }}
  
  else if (nrow(cpg_set_length)==1){ return(cpg_set_length$refseq_mrna) }
  else { return("none") }
  #else { stop() }
})
longest_transcript19 <- unlist(longest_transcript19)
longest_transcript <- c(longest_transcript, longest_transcript19)

longest_transcript20 <- sapply(190001:200000, function(i) {
  
  #print(i)
  cpg_set <- transcript_df[transcript_df$site==unique_cpgs$site[i],]
  #cpg_set_length <- getBM(mart=human_mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_length"))
  
  cpg_set_length <- getBM(mart=mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_start", "transcript_end"))
  cpg_set_length$transcript_length = cpg_set_length$transcript_end - cpg_set_length$transcript_start
  
  if (nrow(cpg_set_length)>1){ max_length_acc = cpg_set_length[cpg_set_length$transcript_length==max(cpg_set_length$transcript_length),"refseq_mrna"]
  if (length(max_length_acc)==1) { return(max_length_acc) }
  else { return(max_length_acc[1]) }}
  
  else if (nrow(cpg_set_length)==1){ return(cpg_set_length$refseq_mrna) }
  else { return("none") }
  #else { stop() }
})
longest_transcript20 <- unlist(longest_transcript20)
longest_transcript <- c(longest_transcript, longest_transcript20)
rm(longest_transcript16, longest_transcript17, longest_transcript18, longest_transcript19, longest_transcript20)

######################################################################

longest_transcript_21 <- sapply(200001:210000, function(i) {
  
  #print(i)
  cpg_set <- transcript_df[transcript_df$site==unique_cpgs$site[i],]
  #cpg_set_length <- getBM(mart=human_mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_length"))
  
  cpg_set_length <- getBM(mart=mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_start", "transcript_end"))
  cpg_set_length$transcript_length = cpg_set_length$transcript_end - cpg_set_length$transcript_start
  
  if (nrow(cpg_set_length)>1){ max_length_acc = cpg_set_length[cpg_set_length$transcript_length==max(cpg_set_length$transcript_length),"refseq_mrna"]
  if (length(max_length_acc)==1) { return(max_length_acc) }
  else { return(max_length_acc[1]) }}
  
  else if (nrow(cpg_set_length)==1){ return(cpg_set_length$refseq_mrna) }
  else { return("none") }
  #else { stop() }
})

longest_transcript_21 <- unlist(longest_transcript_21)
longest_transcript <- c(longest_transcript, longest_transcript_21)

longest_transcript22 <- sapply(210001:220000, function(i) {
  
  #print(i)
  cpg_set <- transcript_df[transcript_df$site==unique_cpgs$site[i],]
  #cpg_set_length <- getBM(mart=human_mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_length"))
  
  cpg_set_length <- getBM(mart=mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_start", "transcript_end"))
  cpg_set_length$transcript_length = cpg_set_length$transcript_end - cpg_set_length$transcript_start
  
  if (nrow(cpg_set_length)>1){ max_length_acc = cpg_set_length[cpg_set_length$transcript_length==max(cpg_set_length$transcript_length),"refseq_mrna"]
  if (length(max_length_acc)==1) { return(max_length_acc) }
  else { return(max_length_acc[1]) }}
  
  else if (nrow(cpg_set_length)==1){ return(cpg_set_length$refseq_mrna) }
  else { return("none") }
  #else { stop() }
})
longest_transcript22 <- unlist(longest_transcript22)
longest_transcript <- c(longest_transcript, longest_transcript22)

longest_transcript23 <- sapply(220001:230000, function(i) {
  
  #print(i)
  cpg_set <- transcript_df[transcript_df$site==unique_cpgs$site[i],]
  #cpg_set_length <- getBM(mart=human_mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_length"))
  
  cpg_set_length <- getBM(mart=mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_start", "transcript_end"))
  cpg_set_length$transcript_length = cpg_set_length$transcript_end - cpg_set_length$transcript_start
  
  if (nrow(cpg_set_length)>1){ max_length_acc = cpg_set_length[cpg_set_length$transcript_length==max(cpg_set_length$transcript_length),"refseq_mrna"]
  if (length(max_length_acc)==1) { return(max_length_acc) }
  else { return(max_length_acc[1]) }}
  
  else if (nrow(cpg_set_length)==1){ return(cpg_set_length$refseq_mrna) }
  else { return("none") }
  #else { stop() }
})
longest_transcript23 <- unlist(longest_transcript23)
longest_transcript <- c(longest_transcript, longest_transcript23)

longest_transcript24 <- sapply(230001:240000, function(i) {
  
  #print(i)
  cpg_set <- transcript_df[transcript_df$site==unique_cpgs$site[i],]
  #cpg_set_length <- getBM(mart=human_mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_length"))
  
  cpg_set_length <- getBM(mart=mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_start", "transcript_end"))
  cpg_set_length$transcript_length = cpg_set_length$transcript_end - cpg_set_length$transcript_start
  
  if (nrow(cpg_set_length)>1){ max_length_acc = cpg_set_length[cpg_set_length$transcript_length==max(cpg_set_length$transcript_length),"refseq_mrna"]
  if (length(max_length_acc)==1) { return(max_length_acc) }
  else { return(max_length_acc[1]) }}
  
  else if (nrow(cpg_set_length)==1){ return(cpg_set_length$refseq_mrna) }
  else { return("none") }
  #else { stop() }
})
longest_transcript24 <- unlist(longest_transcript24)
longest_transcript <- c(longest_transcript, longest_transcript24)

longest_transcript25 <- sapply(240001:250000, function(i) {
  
  #print(i)
  cpg_set <- transcript_df[transcript_df$site==unique_cpgs$site[i],]
  #cpg_set_length <- getBM(mart=human_mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_length"))
  
  cpg_set_length <- getBM(mart=mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_start", "transcript_end"))
  cpg_set_length$transcript_length = cpg_set_length$transcript_end - cpg_set_length$transcript_start
  
  if (nrow(cpg_set_length)>1){ max_length_acc = cpg_set_length[cpg_set_length$transcript_length==max(cpg_set_length$transcript_length),"refseq_mrna"]
  if (length(max_length_acc)==1) { return(max_length_acc) }
  else { return(max_length_acc[1]) }}
  
  else if (nrow(cpg_set_length)==1){ return(cpg_set_length$refseq_mrna) }
  else { return("none") }
  #else { stop() }
})
longest_transcript25 <- unlist(longest_transcript25)
longest_transcript <- c(longest_transcript, longest_transcript25)
rm(longest_transcript_21, longest_transcript22, longest_transcript23, longest_transcript24, longest_transcript25)

longest_transcript26 <- sapply(250001:260000, function(i) {
  
  #print(i)
  cpg_set <- transcript_df[transcript_df$site==unique_cpgs$site[i],]
  #cpg_set_length <- getBM(mart=human_mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_length"))
  
  cpg_set_length <- getBM(mart=mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_start", "transcript_end"))
  cpg_set_length$transcript_length = cpg_set_length$transcript_end - cpg_set_length$transcript_start
  
  if (nrow(cpg_set_length)>1){ max_length_acc = cpg_set_length[cpg_set_length$transcript_length==max(cpg_set_length$transcript_length),"refseq_mrna"]
  if (length(max_length_acc)==1) { return(max_length_acc) }
  else { return(max_length_acc[1]) }}
  
  else if (nrow(cpg_set_length)==1){ return(cpg_set_length$refseq_mrna) }
  else { return("none") }
  #else { stop() }
})
longest_transcript26 <- unlist(longest_transcript26)
longest_transcript <- c(longest_transcript, longest_transcript26)

longest_transcript27 <- sapply(260001:270000, function(i) {
  
  #print(i)
  cpg_set <- transcript_df[transcript_df$site==unique_cpgs$site[i],]
  #cpg_set_length <- getBM(mart=human_mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_length"))
  
  cpg_set_length <- getBM(mart=mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_start", "transcript_end"))
  cpg_set_length$transcript_length = cpg_set_length$transcript_end - cpg_set_length$transcript_start
  
  if (nrow(cpg_set_length)>1){ max_length_acc = cpg_set_length[cpg_set_length$transcript_length==max(cpg_set_length$transcript_length),"refseq_mrna"]
  if (length(max_length_acc)==1) { return(max_length_acc) }
  else { return(max_length_acc[1]) }}
  
  else if (nrow(cpg_set_length)==1){ return(cpg_set_length$refseq_mrna) }
  else { return("none") }
  #else { stop() }
})
longest_transcript27 <- unlist(longest_transcript27)
longest_transcript <- c(longest_transcript, longest_transcript27)

longest_transcript28 <- sapply(270001:280000, function(i) {
  
  #print(i)
  cpg_set <- transcript_df[transcript_df$site==unique_cpgs$site[i],]
  #cpg_set_length <- getBM(mart=human_mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_length"))
  
  cpg_set_length <- getBM(mart=mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_start", "transcript_end"))
  cpg_set_length$transcript_length = cpg_set_length$transcript_end - cpg_set_length$transcript_start
  
  if (nrow(cpg_set_length)>1){ max_length_acc = cpg_set_length[cpg_set_length$transcript_length==max(cpg_set_length$transcript_length),"refseq_mrna"]
  if (length(max_length_acc)==1) { return(max_length_acc) }
  else { return(max_length_acc[1]) }}
  
  else if (nrow(cpg_set_length)==1){ return(cpg_set_length$refseq_mrna) }
  else { return("none") }
  #else { stop() }
})
longest_transcript28 <- unlist(longest_transcript28)
longest_transcript <- c(longest_transcript, longest_transcript28)

longest_transcript29 <- sapply(280001:290000, function(i) {
  
  #print(i)
  cpg_set <- transcript_df[transcript_df$site==unique_cpgs$site[i],]
  #cpg_set_length <- getBM(mart=human_mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_length"))
  
  cpg_set_length <- getBM(mart=mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_start", "transcript_end"))
  cpg_set_length$transcript_length = cpg_set_length$transcript_end - cpg_set_length$transcript_start
  
  if (nrow(cpg_set_length)>1){ max_length_acc = cpg_set_length[cpg_set_length$transcript_length==max(cpg_set_length$transcript_length),"refseq_mrna"]
  if (length(max_length_acc)==1) { return(max_length_acc) }
  else { return(max_length_acc[1]) }}
  
  else if (nrow(cpg_set_length)==1){ return(cpg_set_length$refseq_mrna) }
  else { return("none") }
  #else { stop() }
})
longest_transcript29 <- unlist(longest_transcript29)
longest_transcript <- c(longest_transcript, longest_transcript29)

longest_transcript30 <- sapply(290001:nrow(unique_cpgs), function(i) {
  
  #print(i)
  cpg_set <- transcript_df[transcript_df$site==unique_cpgs$site[i],]
  #cpg_set_length <- getBM(mart=human_mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_length"))
  
  cpg_set_length <- getBM(mart=mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_start", "transcript_end"))
  cpg_set_length$transcript_length = cpg_set_length$transcript_end - cpg_set_length$transcript_start
  
  if (nrow(cpg_set_length)>1){ max_length_acc = cpg_set_length[cpg_set_length$transcript_length==max(cpg_set_length$transcript_length),"refseq_mrna"]
  if (length(max_length_acc)==1) { return(max_length_acc) }
  else { return(max_length_acc[1]) }}
  
  else if (nrow(cpg_set_length)==1){ return(cpg_set_length$refseq_mrna) }
  else { return("none") }
  #else { stop() }
})
longest_transcript30 <- unlist(longest_transcript30)
longest_transcript <- c(longest_transcript, longest_transcript30)
rm(longest_transcript26, longest_transcript27, longest_transcript28, longest_transcript29, longest_transcript30)

######################################################################

#final check to see that all cpg sites have an associated longest transcript annotation
identical(length(longest_transcript), nrow(unique_cpgs)) #TRUE

#put the transcript annotation together in a dataframe with cpg site annotation
unique_cpgs$longest_transcript <- longest_transcript
```

#Gathering remaining CpG sites to be annotated
Next I will perform checks against the original master dataframe ("transcript_df") to see how many CpG sites still need to be annotated after recording all the mRNA transcripts.
```{r outlier sites, eval=F}
#renaming the "unique_cpgs" data frame to "multi_transcript_df", a dataframe containing annotations for sites that have multiple associated transcripts.
multi_transcript_df <- unique_cpgs

#collecting entries that have "none" for the transcript length. These did not have mRNA entries in biomaRt and are thus noncoding RNAs. I will set the "longest_transcript" column value to "NA" for these sites (number to be filled in)
multi_transcript_df[grep("none", multi_transcript_df$longest_transcript),"longest_transcript"] <- NA

#Collecting site-transcript pairs from the master annotation of all transcripts that have not yet been accounted for
#To do this, I will first create a unique identifier for each row of the master annotation, so that the site-transcript pairs can be matched between dataframes
transcript_df$ID <- paste(transcript_df$site, transcript_df$acc, sep=".")

#creating the same format of identifier for the multi_transcript dataframe
multi_transcript_df$ID <- paste(multi_transcript_df$site, multi_transcript_df$longest_transcript, sep=".")

#transcript_df_sub <- transcript_df[transcript_df$ID %in% multi_transcript_df$ID,]

#subsetting the multi_transcript dataframe to site-transcript pairs which were NOT in the original master annotation
#(NOTE TO SELF): these should be all the sites with "none" as their transcript entry. double check this when re-running code
multi_outlier <- multi_transcript_df[-which(multi_transcript_df$ID %in% transcript_df$ID),]
#7,222 CpG sites

#subsetting the master annotation to cpg sites that haven't yet been annotated
transcript_df_outlier <- transcript_df[transcript_df$site %in% multi_outlier$site,]
#20,616 annotations

#further subsetting both the master and multi_transcript dataframes to ncRNA transcripts specifically. This will create the smallest dataframe possible to run the transcript calculation loop on, increasing efficiency
transcript_df_ncrna <- transcript_df_outlier[grep("NR", transcript_df_outlier$acc),] #16,818 annotations
multi_outlier_ncrna <- multi_outlier[multi_outlier$site %in% transcript_df_ncrna$site,] #6,230 sites
```

#Transcript length calculation: noncoding RNAs
I will next perform the same steps as above for CpG sites located in noncoding RNA genes. There are far fewer ncRNA genes, so there is no need to break up the remaining sites into multiple loops and merge later - I will perform this in one loop.
```{r transcript length ncRNA, eval=F}
#annotating ncRNA transcripts from biomaRt
longest_transcript_ncrna <- sapply(1:nrow(multi_outlier_ncrna), function(i) {
  
  print(i)
  cpg_set <- transcript_df_ncrna[transcript_df_ncrna$site==multi_outlier_ncrna$site[i],]
  cpg_set_length <- getBM(mart=mart, filters="refseq_ncrna", values=cpg_set$acc, attributes=c("refseq_ncrna", "transcript_start", "transcript_end"))
  cpg_set_length$transcript_length = cpg_set_length$transcript_end - cpg_set_length$transcript_start
  if (nrow(cpg_set_length)>1){ max_length_acc = cpg_set_length[cpg_set_length$transcript_length==max(cpg_set_length$transcript_length),"refseq_ncrna"]
  if (length(max_length_acc)==1) { return(max_length_acc) }
  else { return(max_length_acc[1]) }}
  
  else if (nrow(cpg_set_length)==1){ return(cpg_set_length$refseq_ncrna) }
  else { return("none") }
  #else { stop() }
})

#putting the longest transcript annotation back into the subsetted ncRNA dataframe
longest_transcript_ncrna <- unlist(longest_transcript_ncrna)
multi_outlier_ncrna$longest_transcript <- longest_transcript_ncrna

#putting the ncRNA annotations into the larger multi_transcript_df dataframe
multi_transcript_df$longest_transcript <- as.character(multi_transcript_df$longest_transcript)
multi_transcript_df[multi_transcript_df$site %in% multi_outlier_ncrna$site,"longest_transcript"] <- longest_transcript_ncrna
```

#Taking care of remaining outlier CpG sites
Even after annotating the ncRNA genes, there were still some CpG sites left without a transcript annotation. Some of these have annotations in hg38 - I queried hg38 next to try and fill in remaining gaps.
```{r outliers 2, eval=F}
#Collecting "outlier" sites that still need to be annotated
multi_transcript_df[grep("none", multi_transcript_df$longest_transcript),"longest_transcript"] <- NA
multi_transcript_df$ID <- paste(multi_transcript_df$site, multi_transcript_df$longest_transcript, sep=".")
nrow(multi_outlier <- multi_transcript_df[-which(multi_transcript_df$ID %in% transcript_df$ID),]) #2984
nrow(transcript_df_outlier <- transcript_df[transcript_df$site %in% multi_outlier$site,]) #8291

#the last annotations aren't in the hg19 biomart. use hg38 biomart
mart <- useMart("ENSEMBL_MART_ENSEMBL", dataset="hsapiens_gene_ensembl", host="http://Aug2017.archive.ensembl.org")

#getting ncRNA transcript annotations
longest_transcript_ncrna_outlier <- sapply(1:nrow(multi_outlier), function(i) {
  
  print(i)
  cpg_set <- transcript_df_outlier[transcript_df_outlier$site==multi_outlier$site[i],]
  cpg_set_length <- getBM(mart=mart, filters="refseq_ncrna", values=cpg_set$acc, attributes=c("refseq_ncrna", "transcript_start", "transcript_end"))
  cpg_set_length$transcript_length = cpg_set_length$transcript_end - cpg_set_length$transcript_start
  if (nrow(cpg_set_length)>1){ max_length_acc = cpg_set_length[cpg_set_length$transcript_length==max(cpg_set_length$transcript_length),"refseq_ncrna"]
  if (length(max_length_acc)==1) { return(max_length_acc) }
  else { return(max_length_acc[1]) }}
  
  else if (nrow(cpg_set_length)==1){ return(cpg_set_length$refseq_ncrna) }
  else { return("none") }
  #else { stop() }
})

longest_transcript_ncrna_outlier <- unlist(longest_transcript_ncrna_outlier)
multi_outlier$longest_transcript <- longest_transcript_ncrna_outlier
multi_outlier$longest_transcript <- as.character(multi_outlier$longest_transcript)
multi_transcript_df[multi_transcript_df$site %in% multi_outlier$site,"longest_transcript"] <- longest_transcript_ncrna_outlier
multi_transcript_df[grep("none", multi_transcript_df$longest_transcript),"longest_transcript"] <- NA
multi_transcript_df$ID <- paste(multi_transcript_df$site, multi_transcript_df$longest_transcript, sep=".")
nrow(multi_outlier <- multi_transcript_df[-which(multi_transcript_df$ID %in% transcript_df$ID),]) #1,289 sites
nrow(transcript_df_outlier <- transcript_df[transcript_df$site %in% multi_outlier$site,]) #3,734 annotations

#getting mRNA transcript annotations
longest_transcript_mrna_outlier <- sapply(1:nrow(multi_outlier), function(i) {
  
  print(i)
  cpg_set <- transcript_df_outlier[transcript_df_outlier$site==multi_outlier$site[i],]
  cpg_set_length <- getBM(mart=mart, filters="refseq_mrna", values=cpg_set$acc, attributes=c("refseq_mrna", "transcript_start", "transcript_end"))
  cpg_set_length$transcript_length = cpg_set_length$transcript_end - cpg_set_length$transcript_start
  if (nrow(cpg_set_length)>1){ max_length_acc = cpg_set_length[cpg_set_length$transcript_length==max(cpg_set_length$transcript_length),"refseq_mrna"]
  if (length(max_length_acc)==1) { return(max_length_acc) }
  else { return(max_length_acc[1]) }}
  
  else if (nrow(cpg_set_length)==1){ return(cpg_set_length$refseq_mrna) }
  else { return("none") }
  #else { stop() }
})
length(unique(longest_transcript_mrna_outlier <- unlist(longest_transcript_mrna_outlier))) #165 (additional transcripts present in the hg38 annotation)
multi_outlier$longest_transcript <- longest_transcript_mrna_outlier
multi_outlier$longest_transcript <- as.character(multi_outlier$longest_transcript)
multi_transcript_df[multi_transcript_df$site %in% multi_outlier$site,"longest_transcript"] <- longest_transcript_mrna_outlier
```

#Final checks and saving
Checking against the transcript_df annotation, there were still more un-annotated CpG sites remaining. These transcripts were not in either biomaRt database because they have been removed from NCBI due to insufficient supporting evidence. I decided to leave the annotation for these in my multi_transcript dataframe as "none" in light of this.
```{r final check, eval=F}
#Collecting outlier CpG sites
multi_transcript_df[grep("none", multi_transcript_df$longest_transcript),"longest_transcript"] <- NA
multi_transcript_df$ID <- paste(multi_transcript_df$site, multi_transcript_df$longest_transcript, sep=".")
multi_outlier <- multi_transcript_df[-which(multi_transcript_df$ID %in% transcript_df$ID),]
transcript_df_outlier <- transcript_df[transcript_df$site %in% multi_outlier$site,]

#910 entries remain (160 unique transcripts).
#364 entries remain. These represent transcripts that were removed from NCBI due to insuffient supporting evidence.

write.table(multi_transcript_df, file="~/KoborLab/kobor_space/sschaffner/LUHMES_201806/clean_scripts/multi_transcript_df_202008.txt", sep="\t")
```